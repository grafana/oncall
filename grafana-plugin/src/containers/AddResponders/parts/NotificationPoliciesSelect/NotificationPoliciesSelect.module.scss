.select {
  width: 150px !important;
}
