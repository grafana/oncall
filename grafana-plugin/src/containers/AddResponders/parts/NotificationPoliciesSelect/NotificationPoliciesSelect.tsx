import React, { FC } from 'react';

import { SelectableValue } from '@grafana/data';
import { Select, ActionMeta } from '@grafana/ui';
import cn from 'classnames/bind';

import { NotificationPolicyValue } from 'containers/AddResponders/AddResponders.types';

import styles from './NotificationPoliciesSelect.module.scss';

const cx = cn.bind(styles);

type Props = {
  disabled?: boolean;
  important: boolean;
  onChange: (value: SelectableValue<number>, actionMeta: ActionMeta) => void;
};

export const NotificationPoliciesSelect: FC<Props> = ({ disabled = false, important, onChange }) => (
  <Select
    className={cx('select')}
    width="auto"
    isSearchable={false}
    value={Number(important)}
    options={[
      {
        value: NotificationPolicyValue.Default,
        label: 'Default',
        description: 'Use "Default notifications" from users personal settings',
      },
      {
        value: NotificationPolicyValue.Important,
        label: 'Important',
        description: 'Use "Important notifications" from users personal settings',
      },
    ]}
    onChange={onChange}
    disabled={disabled}
  />
);
