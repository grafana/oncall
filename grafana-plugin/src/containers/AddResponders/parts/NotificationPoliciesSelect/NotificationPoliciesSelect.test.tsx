import React from 'react';

import { render } from '@testing-library/react';

import { NotificationPoliciesSelect } from './NotificationPoliciesSelect';

describe('NotificationPoliciesSelect', () => {
  test('it renders properly', () => {
    const component = render(<NotificationPoliciesSelect important={false} onChange={() => {}} />);
    expect(component.container).toMatchSnapshot();
  });

  test('disabled state', async () => {
    const component = render(<NotificationPoliciesSelect disabled important={false} onChange={() => {}} />);
    expect(component.container).toMatchSnapshot();
  });
});
