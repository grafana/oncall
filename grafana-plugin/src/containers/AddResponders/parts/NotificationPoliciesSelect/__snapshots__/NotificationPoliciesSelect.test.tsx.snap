// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPoliciesSelect disabled state 1`] = `
<div>
  <div
    class="css-e49k3t-input-wrapper select css-8k5qe3-SelectContainer"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-3-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="css-1i88p6p"
    >
      <div
        class="css-1q0c0d5-grafana-select-value-container"
      >
        <div
          class="css-sr1xkh-singleValue css-upz218-SingleValue"
        >
          Default
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-haspopup="true"
          aria-readonly="true"
          class="css-mohuvp-dummyInput-DummyInput"
          disabled=""
          id="react-select-3-input"
          inputmode="none"
          role="combobox"
          tabindex="0"
          value=""
        />
      </div>
      <div
        class="css-zyjsuv-input-suffix"
      >
        <div
          class="css-1j2891d-Icon"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationPoliciesSelect it renders properly 1`] = `
<div>
  <div
    class="css-1nmqu8c-input-wrapper select css-8k5qe3-SelectContainer"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="css-1i88p6p"
    >
      <div
        class="css-1q0c0d5-grafana-select-value-container"
      >
        <div
          class="css-8nwx1l-singleValue css-upz218-SingleValue"
        >
          Default
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-haspopup="true"
          aria-readonly="true"
          class="css-mohuvp-dummyInput-DummyInput"
          id="react-select-2-input"
          inputmode="none"
          role="combobox"
          tabindex="0"
          value=""
        />
      </div>
      <div
        class="css-zyjsuv-input-suffix"
      >
        <div
          class="css-1j2891d-Icon"
        />
      </div>
    </div>
  </div>
</div>
`;
